src/libraries/Common/src/System/Reflection/AssemblyNameParser.netstandard.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+
+#nullable enable
+
+namespace System.Reflection
+{
+    internal ref partial struct AssemblyNameParser
+    {
+        private static HashSet<string>? _predefinedCultureNames;
+        private static readonly object _predefinedCultureNamesLock = new object();
+
+        private static bool IsPredefinedCulture(string cultureName)
```

Comment:
Why is this in its own file for netstandard but the net8.0 one is in the main AssemblyNameParser.cs file ifdef'd?

Reply:
The net8.0 implementation was very small compared to .NET Standard. Should I move the `net8.0` implementation to a dedicated file too?

Reply:
Personally I think it'd be nice to keep it symmetric.

Reply:
This is only needed for the "strict" mode. I am not convinced that we are going to need this method at the end. I do not think that the logic in this method belongs to System.Reflection.Metadata.

Reply:
For now I am going to remove it as it touches the "strict" mode.

> I do not think that the logic in this method belongs to System.Reflection.Metadata.

This was my initial assumption as well,  mostly because the serialization binder accepts `typeName` and `assemblyName`:

https://github.com/dotnet/runtime/blob/bf828d87a7cfdff5c4c7f219ef77b6b4c0be04cf/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/SerializationBinder.cs#L14

But then I realized, that for generic type arguments, the type names contain assembly names:

So for `List<int>` we would get following values (I am skipping type forwarding for brevity):

```log
typeName = "System.Collections.Generic.List`1[[System.Int32, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]"
assemblyName = "System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e"
```

And in such case, it would be very hard to reject hostile assembly names from outside of the parser (we would need to write another parser that would parse the type names just to search for assembly names?). We also can't just reject already-parsed AssemblyNames, as they can instantiate cultures at the time of parsing.

cc @GrabYourPitchforks